internal/mode/static/state/dataplane/configuration.go

```diff
@@ -41,24 +41,28 @@ func BuildConfiguration(
 		return Configuration{Version: configVersion}
 	}
 
-	upstreams := buildUpstreams(ctx, g.Gateway.Listeners, resolver)
+	upstreams := buildUpstreams(ctx, g.Gateway.Listeners, serviceResolver)
 	httpServers, sslServers := buildServers(g, generator)
 	backendGroups := buildBackendGroups(append(httpServers, sslServers...))
 	keyPairs := buildSSLKeyPairs(g.ReferencedSecrets, g.Gateway.Listeners)
 	certBundles := buildCertBundles(g.ReferencedCaCertConfigMaps, backendGroups)
 	telemetry := buildTelemetry(g)
 	baseHTTPConfig := buildBaseHTTPConfig(g)
+	var tlsServers []Layer4VirtualServer
```

Comment:
You don't need to declare these variables. Instead, you can omit the `TLSPassthroughServers` and `StreamUpstreams` fields from the config. If a field is omitted from a struct, the zero value for that field is used.